receiver/prometheusreceiver/internal/prom_to_otlp.go

```diff
@@ -21,28 +21,36 @@ import (
 	conventions "go.opentelemetry.io/collector/model/semconv/v1.5.0"
 )
 
-const (
-	localHostIPv4 = "127.0.0.1"
-	anyIPIPv4     = "0.0.0.0"
-)
+// isAllowedHost returns if a host can be used as a value for the 'host.name' key.
+// localhost-like hosts and unspecified (0.0.0.0) hosts are not allowed.
+func isAllowedHost(host string) bool {
```

Comment:
Nitpicking: `isDiscernibleHost(host)` would sound clearer to me than `isAllowedHost(host)`, which makes me think of ACLs, firewalls, etc. 
(totally optional suggestion)

Reply:
Done!